test/unit/alphabet/aminoacid/aminoacid_test_template.hpp

```diff
@@ -16,15 +16,24 @@
 using namespace seqan3;
 
 template <typename T>
-class aminoacid : public ::testing::Test
-{};
+using aminoacid = ::testing::Test;
 
 TYPED_TEST_CASE_P(aminoacid);
 
 TYPED_TEST_P(aminoacid, concept_check)
 {
     EXPECT_TRUE(AminoacidAlphabet<TypeParam>);
     EXPECT_TRUE(AminoacidAlphabet<TypeParam &>);
+    EXPECT_TRUE(AminoacidAlphabet<TypeParam const>);
+    EXPECT_TRUE(AminoacidAlphabet<TypeParam const &>);
+}
+
+TYPED_TEST_P(aminoacid, type_properties)
+{
+    EXPECT_TRUE((std::is_trivially_copyable_v<TypeParam>));
+    EXPECT_FALSE((std::is_trivially_default_constructible_v<TypeParam>));
+    EXPECT_FALSE((std::is_trivial_v<TypeParam>));
+    EXPECT_TRUE((std::is_standard_layout_v<TypeParam>));
```

Comment:
Hm... I think we could just add the following to the template:
```cpp
    EXPECT_TRUE((std::is_trivially_copyable_v<TypeParam>));
    EXPECT_TRUE((std::is_standard_layout_v<TypeParam>));
```

Why should we enforce that they are not trivially-default-constructible?

Reply:
I don't want to enforce that, but I want to know what types have what properties and most are not trivial. If some type changes to trivial, it would be better.

Reply:
>  I want to know what types have what properties

But which properties do you mean? In the documentation we state that:

```
It is highly recommended that non-reference types that model this concept, also model:
    std::Regular
    std::is_trivially_copyable
    seqan3::StandardLayout
```

So it might make sense to check those concepts. Regular and StandardLayout can be checked in the alphabet_test_template. trivially_copyable also except that alphabet_proxy needs an exception (I am not sure if there even is a test for alphabet_proxy inside test_template so maybe it doesn't even need an exception). 

But why do you want to check trivial? No code depends on this, it doesn't influence anything.

> If some type changes to trivial,

How would a type "change"? Why does it matter? Also I can promise that with the current design, anything that uses aphabet_base cannot be trivial ever.